include/envoy/common/platform.h

```diff
@@ -248,19 +248,12 @@ struct mmsghdr {
 #endif // __ANDROID_API__ < 24
 #endif // ifdef __ANDROID_API__
 
-#ifdef __linux__
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif
-
 // https://android.googlesource.com/platform/bionic/+/master/docs/status.md
 // ``pthread_getname_np`` is introduced in API 26
-#ifdef __ANDROID_API__
-#if __ANDROID_API__ > 26
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif // __ANDROID_API__ > 26
-#endif // ifdef __ANDROID_API__
-
-// Ensure `SUPPORTS_PTHREAD_GETNAME_NP` is set
-#ifndef SUPPORTS_PTHREAD_GETNAME_NP
-#define SUPPORTS_PTHREAD_GETNAME_NP 0
-#endif
+#if defined(__ANDROID_API__)
+#if __ANDROID_API__ >= 26
+#define SUPPORTS_PTHREAD_GETNAME_NP
+#endif // __ANDROID_API__ >= 26
+#elif defined(__linux__)
+#define SUPPORTS_PTHREAD_GETNAME_NP
+#endif // defined(__ANDROID_API__)
```

Comment:
```suggestion
#define SUPPORTS_PTHREAD_NAMING 0

#ifdef __linux__
#define SUPPORTS_PTHREAD_NAMING 1
#endif

#if defined(__ANDROID_API__) && __ANDROID_API__ < 26
#define SUPPORTS_PTHREAD_NAMING 0
#endif
```

Reply:
If we do this the compiler will complain that we're redefining `SUPPORTS_PTHREAD_NAMING`.

```
bazel-out/android-x86-fastbuild/bin/external/envoy/include/envoy/common/_virtual_includes/base_includes/envoy/common/platform.h:255:9: error: 'SUPPORTS_PTHREAD_NAMING' macro redefined [-Werror,-Wmacro-redefined]
#define SUPPORTS_PTHREAD_NAMING 1
        ^
bazel-out/android-x86-fastbuild/bin/external/envoy/include/envoy/common/_virtual_includes/base_includes/envoy/common/platform.h:253:9: note: previous definition is here
#define SUPPORTS_PTHREAD_NAMING 0
        ^
bazel-out/android-x86-fastbuild/bin/external/envoy/include/envoy/common/_virtual_includes/base_includes/envoy/common/platform.h:258:9: error: 'SUPPORTS_PTHREAD_NAMING' macro redefined [-Werror,-Wmacro-redefined]
#define SUPPORTS_PTHREAD_NAMING 0
        ^
bazel-out/android-x86-fastbuild/bin/external/envoy/include/envoy/common/_virtual_includes/base_includes/envoy/common/platform.h:255:9: note: previous definition is here
#define SUPPORTS_PTHREAD_NAMING 1
        ^
2 errors generated.
```

Reply:
Ah, we're being strict with `-Wmacro-redfined`.

I would still suggest simplifying and renaming.